google/cloud/sql/connector/utils.py

```diff
@@ -56,19 +68,37 @@ def generate_keys():
     key_size - The cryptography documentation recommended a key_size
     of at least 2048.
     """
-    private_key_obj = rsa.generate_private_key(
+    return rsa.generate_private_key(
         backend=default_backend(), public_exponent=65537, key_size=2048
     )
-    private_key = private_key_obj.private_bytes(
+
+
+@run_function_as_async
+def get_private_key_bytes(private_key_obj):
+    """Helper function to get private key bytes from private key object."""
+    return private_key_obj.private_bytes(
         encoding=serialization.Encoding.PEM,
         format=serialization.PrivateFormat.TraditionalOpenSSL,
         encryption_algorithm=serialization.NoEncryption(),
     )
-    public_key = private_key_obj.public_key().public_bytes(
+
+
+@run_function_as_async
+def get_public_key_bytes(private_key_obj):
+    """Helper function to get public key bytes from private key object."""
+    return private_key_obj.public_key().public_bytes(
         encoding=serialization.Encoding.PEM,
         format=serialization.PublicFormat.SubjectPublicKeyInfo,
     )
```

Comment:
I'm a little concern that splitting these up isn't adding a lot of value - they are mostly just formatting, and not sure the overhead of using a background thread will outweigh the cost to format.